scene/resources/2d/tile_set.cpp

```diff
@@ -4061,7 +4075,9 @@ bool TileSet::_get(const StringName &p_name, Variant &r_ret) const {
 			r_ret = get_occlusion_layer_sdf_collision(index);
 			return true;
 		}
-	} else if (components.size() == 2 && components[0].begins_with("physics_layer_") && components[0].trim_prefix("physics_layer_").is_valid_int()) {
+	} else
+#ifndef PHYSICS_2D_DISABLED
+			if (components.size() == 2 && components[0].begins_with("physics_layer_") && components[0].trim_prefix("physics_layer_").is_valid_int()) {
```

Comment:
Wouldn't this work just as well and avoid having to indent a huge chunk of code?
I haven't reviewed in depth but depending on what comes later this stray `} else` instead of `} else if` may also have the potential to break some code (it will always be entered, unlike `} else if` if the condition is false).
```suggestion
#ifndef PHYSICS_2D_DISABLED
	} else if (components.size() == 2 && components[0].begins_with("physics_layer_") && components[0].trim_prefix("physics_layer_").is_valid_int()) {
```